spec/dummy/app/controllers/application_controller.rb

```diff
@@ -1,9 +1,17 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
 
+  content_security_policy only: :strict_csp do |policy|
+    policy.script_src :self, :strict_dynamic
+  end
+
   def root
   end
 
+  def strict_csp
```

Comment:
Where this controller action is being used?

Reply:
Ah, good catch! I needed this action to duplicate the actual problem (JS errors), but I guess I hadn't added a test that used it yet.

Since `spec/xray/middleware_spec.rb` (where it was) only checks implementation details (that a certain `nonce` attribute is added) but doesn't actually test that this is enough to prevent JS/CSP errors, I added a quick end-to-end (Webdriver) test that verifies this in a real browser. Seems to work...